exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/Metadata.java

```diff
@@ -149,20 +157,25 @@ public static ParquetTableMetadata_v3 getParquetTableMetadata(Map<FileStatus, Fi
    * Get the parquet metadata for the table by reading the metadata file
    *
    * @param fs current file system
-   * @param path The path to the metadata file, located in the directory that contains the parquet files
+   * @param paths The path to the metadata file, located in the directory that contains the parquet files
    * @param metaContext metadata context
    * @param readerConfig parquet reader configuration
    * @return parquet table metadata. Null if metadata cache is missing, unsupported or corrupted
    */
   public static @Nullable ParquetTableMetadataBase readBlockMeta(FileSystem fs,
-                                                                 Path path,
+                                                                 List<Path> paths,
                                                                  MetadataContext metaContext,
                                                                  ParquetReaderConfig readerConfig) {
-    if (ignoreReadingMetadata(metaContext, path)) {
-      return null;
-    }
     Metadata metadata = new Metadata(readerConfig);
-    metadata.readBlockMeta(path, false, metaContext, fs);
+    if (paths.isEmpty()) {
+      metaContext.setMetadataCacheCorrupted(true);
+    }
+    for (Path path: paths) {
+      if (ignoreReadingMetadata(metaContext, path)) {
+        return null;
```

Comment:
Paths contain paths for Summary and File metadata file. If we pull out the ignoreReadingMetadata out of the for loop, we won't be able to ignore reading the Paths[1] file if the Paths[0] one fails. We would like to ignore reading of the metadata if the reading of the Paths[0] file fails.

Reply:
I see that `ignoreReadingMetadata()` method does not actually examine the `path`.  It just checks if the metadata cache is corrupted.  From a logical standpoint, I don't see why one would ignore the Summary file but not the File metadata (or vice-versa).   From a performance reason also, suppose paths[0] was not ignored (hypothetically), and we execute the next statement `readBlockMeta()` and read the entire File metadata and subsequently find out that paths[1] is ignored, it would be wasted effort.  What do you think ?

Reply:
Lets take this scenario where the summary file is corrupted but the file metadata file is intact. If we pull out the ignoreReadingMetadata (), we do this check of ignoring only before reading the summary file. But after reading the corrupted summary file, metadatacachecorrupted will be set to true. Reading of filemetadata will not be skipped if we don't check metadatacachecorrupted status. Regarding the performance  if the Paths[1] is corrupt, we will get to know that it is corrupt only in the readBlockMeta(). So even pulling out ignoreReadingMetadata() doesn't help this case. And always Paths[0] has summary and since summary is not very big it wouldn't impact the performance a lot.

Reply:
Pls add a comment indicating that the ignore flag is set as part of `readBlockMeta()` which is called later in this loop, so each iteration of the loop needs to check if a previous call to readBlockMeta found a corrupted metadata.

Reply:
Added this comment.